File: C#/Container.Tests/ContainerTestClasses/MultipleDependenciesAndInterface.cs
using IoC.Tests.ContainerTestClasses.Mock;

namespace IoC.Tests.ContainerTestClasses
{
    class MultipleDependenciesAndInterface
    {
        public IMockService InterfaceService { get; set; }
        public IMockService Service1 { get; set; }
        public IMockService Service2 { get; set; }

        public MultipleDependenciesAndInterface(IMockService iService, MockService1 service1, MockService2 service2)
        {
            InterfaceService = iService;
            Service1 = service1;
            Service2 = service2;
        }
    }
}

File: Java/src/test/java/Mock/MultipleDependenciesAndInterface.java
package Mock;

/**
 * Created by user on 22/9/2015.
 */
public class MultipleDependenciesAndInterface {
    public IMockService interfaceService ;
    public IMockService service1 ;
    public IMockService service2 ;

    public MultipleDependenciesAndInterface(IMockService iService, MockService1 service1, MockService2 service2)
    {
        this.interfaceService = iService;
        this.service1 = service1;
        this.service2 = service2;
    }
}
